include/mbgl/mtl/buffer_resource.hpp

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <mbgl/gfx/vertex_buffer.hpp>
+#include <mbgl/mtl/mtl_fwd.hpp>
+
+#include <Foundation/NSSharedPtr.hpp>
+#include <Metal/MTLBuffer.hpp>
+#include <Metal/MTLDevice.hpp>
+
+namespace mbgl {
+namespace mtl {
+
+class BufferResource {
+public:
+    BufferResource() = default;
+    BufferResource(MTLDevicePtr, const void*, std::size_t size, NS::UInteger usage);
```

Comment:
Is the `usage` parameter obvious to someone familiar with the Metal API?

Otherwise, consider documenting this parameter.

Reply:
Good catch, that's the wrong type anyway.